src/math/types/seconds_per_liquidity.rs

```diff
@@ -46,7 +46,7 @@ impl SecondsPerLiquidity {
 
         Ok(Self::new(
             Self::checked_from_value(
-                U128::from(delta_time)
+                U256::from(delta_time)
```

Comment:
Okay, so the result is inside U128, but this has failed because of casting liquidity to U128. This means that the maximum intermediate is `L_max = 2^256 - 1`.

This remains solution or solution based on increasing the type just before the required div:
```Rust
        Ok(Self::new(
            Self::checked_from_value(
                Self::from_value::<U256, U128>(
                    U128::from(delta_time)
                        .checked_mul(Self::one().cast())
                        .ok_or_else(|| err!(TrackableError::MUL))?
                        .checked_mul(Liquidity::one().cast())
                        .ok_or_else(|| err!(TrackableError::MUL))?,
                )
                .checked_div(liquidity.get())
                .ok_or_else(|| err!(TrackableError::DIV))?,
            )
            .map_err(|_| err!(TrackableError::cast::<U128>().as_str()))?,
        ))
```